src/jdk.accessibility/windows/native/jaccesswalker/jaccesswalker.cpp

```diff
@@ -543,8 +543,8 @@ void Jaccesswalker::addComponentNodes(long vmID, AccessibleContext context,
     } else {
         char s[LINE_BUFSIZE];
         snprintf( s, sizeof(s),
-            "ERROR calling GetAccessibleContextInfo; vmID = %X, context = %p",
-            vmID, (void*)context );
+            "ERROR calling GetAccessibleContextInfo; vmID = %lX, context = %p",
+            reinterpret_cast<unsigned long>(vmID), (void*)context );
```

Comment:
do you need this cast? I checked a few compilers and passing a signed long to "%lX" was fine with them.

Reply:
gcc will crash with a warning about a mismatched format specifier between signed and unsigned if this isn't done, unfortunately

Reply:
Which gcc? This code compiles without warnings:
```
#include <stdio.h>
int main() {
	unsigned long i = 1;
	long j = 2;
	printf("%ld %ld %lx %lx %lu %lu\n", i, j, i, j, i, j);
	return 0;
}
```
```
# gcc -Wall -Wextra -Wformat=2 test.c
# gcc --version
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Reply:
I'm currently running version 13.1, win32 threads

Reply:
I'll retry again, maybe the warning has changed now

Reply:
Seems like it doesn't trigger any longer, I'll revert the cast. Thanks for catching this